kpatch-build/kpatch-build

```diff
@@ -642,6 +641,20 @@ fi
 
 trace_on
 
+if [[ -n "${VMLINUX}" || -f "$BASE/vmlinux" ]]; then
+	ARCH="$(file "${VMLINUX:-$BASE/vmlinux}")"
+else
+	ARCH="$(uname -m)"
+fi
+
+# normalize to kernel's $ARCH preferences
+case "${ARCH}" in
+(*x86?64*) ARCH=x86_64;;
+(*ppc64le*) ARCH=ppc64le;;
+(*aarch64*) ARCH=arm64;;
+(*) die "Kernel ${VMLINUX:-$BASE/vmlinux} has Unsupported Architecture (${ARCH})";;
+esac
+
```

Comment:
exactly - I'll split off as a separate PR, to be discussed and resolved as an orthogonal issue.
It was bundled in here because @t-msn & I reworked @surajjs95 changes from '#ifdef __arch64__' to 'switch(kelf->arch)', and this extended the cross-arch transformation

Reply:
but when this comes back as cross-architecture support proposal, it will use elf/uname arch names like aarch64, not try to subvert kpatch-build tradition by changing the arch names to kernel makefile versions like arm64+x86_64